src/Gui/DlgMacroExecuteImp.cpp

```diff
@@ -60,14 +60,30 @@ namespace Dialog
 class MacroItem: public QTreeWidgetItem
 {
 public:
-    MacroItem(QTreeWidget* widget, bool systemwide)
+    MacroItem(QTreeWidget* widget, bool systemwide, const QString& dirPath)
         : QTreeWidgetItem(widget)
         , systemWide(systemwide)
+        , dirPath(dirPath)
     {}
 
+    /**
+     * Override setText method to additionally set toolTip with text of absolute
+     * file path. There may be different macros with same names from different
+     * system paths. So it could be helpful for user to show where exactly macro
+     * is placed.
+     */
+    void setText(int column, const QString& text)

```

Comment:
Because QTreeWidgetItem::setText is not virtual and thus avoid any confusion it's better to rename (e..g. setFileName()) this method. There you can also drop the column parameter.